src/DittoMachine.sol

```diff
@@ -55,10 +55,21 @@ contract DittoMachine is ERC721, ERC721TokenReceiver, ERC1155TokenReceiver {
         uint256 term;
     }
 
+    //protoId is a precursor hash to a cloneId used to identify tokenId/erc20 pairs
+    mapping(uint256 => uint256) public protoIdToIndexHead;
+    // mapping to track the next index after the index head
+    mapping(uint256 => mapping(uint256 => uint256)) public protoIdToIndexToPrior;
+    // mapping to track the index before the index head
+    mapping(uint256 => mapping(uint256 => uint256)) public protoIdToIndexToAfter;
```

Comment:
should this be renamed `protoIdToIndexBefore`?

Reply:
We can if you'd like, I used `Prior` because its the same length as `After` and my monkey brain found it visually nice to look as if they're same length lol.

Reply:
ignore.

Reply:
I think I got confused with the comment saying `after` and before` and the variable names seemingly opposite to that. I'll probably understand the reason once I go through the entire implementation.

Reply:
Ok, do you think the comment and variable name mismatch? like the comment for `protoIdToIndexToPrior` should be `    

```
// mapping to track the next index prior the index head
```

Reply:
I actually don't think it should be mentioning the index head at all in these comments now that i think about it. like lets imagine there are three clones minted at these indexes `0, 1, 2
`
each index has an after pointer like this:
```
0 -> 1 -> 2 -> 3
```
and each index should have prior pointers like this:
```
0 <- 1 <- 2 <- 3
```

Reply:
will edit comments to clarify this